doc/source/whatsnew/v1.2.1.rst

```diff
@@ -15,6 +15,7 @@ including other versions of pandas.
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
 - The deprecated attributes ``_AXIS_NAMES`` and ``_AXIS_NUMBERS`` of :class:`DataFrame` and :class:`Series` will no longer show up in ``dir`` or ``inspect.getmembers`` calls (:issue:`38740`)
+- Bug in float-like strings having trailing 0's truncated (:issue:`38708`)
```

Comment:
Can you add in "repr of" and "after the decimal" here for clarity that it is just a bug in regards to presentation (which is still important!).

Reply:
these are not floats, rather strings in an `object` dtype

Reply:
Done